lib/fs/filesystem_test.go

```diff
@@ -41,3 +41,68 @@ func TestIsInternal(t *testing.T) {
 		}
 	}
 }
+
+func TestCanonicalize(t *testing.T) {
+	type testcase struct {
+		path     string
+		expected string
+		ok       bool
+	}
+	cases := []testcase{
+		// Valid cases
+		{"/bar", "bar", true},
```

Comment:
You have the same test case a bunch of times in this table, here and below. I guess it comes from search/replace on another table? Just remove duplicates and it's cool.